src/Build/Graph/GraphBuilder.cs

```diff
@@ -495,24 +496,55 @@ void AddGraphBuildGlobalVariable(PropertyDictionary<ProjectPropertyInstance> glo
             }
         }
 
-        private ParsedProject ParseProject(ConfigurationMetadata configurationMetadata)
+        private ParsedProject ParseProject(ConfigurationMetadata configurationMetadata, bool enableDynamicPlatformResolution)
         {
             // TODO: ProjectInstance just converts the dictionary back to a PropertyDictionary, so find a way to directly provide it.
             var globalProperties = configurationMetadata.GlobalProperties.ToDictionary();
-
-            var projectInstance = _projectInstanceFactory(
-                configurationMetadata.ProjectFullPath,
-                globalProperties,
-                _projectCollection);
+            ProjectGraphNode graphNode;
+            ProjectInstance projectInstance;
+           
+            if (!enableDynamicPlatformResolution)
+            {
+                projectInstance = _projectInstanceFactory(
+                                    configurationMetadata.ProjectFullPath,
+                                    globalProperties,
+                                    _projectCollection);
+            }
+            else
+            {
+                projectInstance = _projectInstanceFactory(
+                                    configurationMetadata.ProjectFullPath,
+                                    null,
+                                    _projectCollection);
+            }
 
             if (projectInstance == null)
             {
                 throw new InvalidOperationException(ResourceUtilities.GetResourceString("NullReferenceFromProjectInstanceFactory"));
             }
 
-            var graphNode = new ProjectGraphNode(projectInstance);
+            if (enableDynamicPlatformResolution)
+            {
+                var SelectedPlatform = PlatformNegotiation.GetNearestPlatform(projectInstance.GetPropertyValue(PlatformMetadataName), projectInstance.GetPropertyValue(PlatformsMetadataName), projectInstance.GetPropertyValue(PlatformLookupTableMetadataName), configurationMetadata.PreviousPlatformLookupTable, projectInstance.FullPath, configurationMetadata.PreviousPlatform);
+
+                if (SelectedPlatform != null)
+                {
+                    globalProperties[PlatformMetadataName] = SelectedPlatform;
+                    if (SelectedPlatform.Equals(String.Empty))
+                    {
+                        globalProperties.Remove(PlatformMetadataName);
+                    }
+                }
```

Comment:
the problem here is that the global platform property would not be removed from the previous node. since platform is passed down from referencer to reference if we don't specifically remove or replace it will stay the same as whatever the referencer built as